### DISCUSSION TITLE
How can multi-blogs use different  Bloglayout?

### DISCUSSION THREAD

**Discussion by ArchFeh** [facebook/docusaurus#10269] [Q&A] [answered]
Hello everyone, I made multi-blogs, and eject ``BlogLayout``. I want in Blog A show a banner with "This is Blog A", and in Blog B shows " This is Blog B", how can I do this? I can just made one same banner in ``BlogLayout`` for  the both blog pages. Or is there any API to get plugin ID？

**Comment by slorber** [comment, marked as answer]
Not documented but you could use the useRouteContext() hook that we use internally

https://github.com/facebook/docusaurus/pull/6933

It gives you the plugin name/id that created a given route

**Reply by ArchFeh** [reply]
Thank u for information！

**Comment by Josh-Cena** [comment]
Btw if you really want to have different layout implementations that's more significant than a single condition, the blog plugin allows you to configure the top-level components for each type of page. See https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-blog#blogListComponent
